docs/uplink/rest-api.md

```diff
@@ -28,76 +48,147 @@ export TOKEN=$(kubectl get secret -n inlets client-api-token \
   | base64 --decode)
 ```
 
+Use the token as bearer token in the `Authorization` header when making requests to the API.
+
+### OAuth
+
+You should be able to configure any OpenID Connect (OIDC) compatible identity provider for use with Inlets Uplink.
+
+1. Register a new client (application) for Inlets Uplink with your identity provider.
+2. Enable the required authentication flows.
+  The Client Credentials flow is ideal for serve-to-server interactions where there is no direct user involvement. This is the flow we recommend and use in our examples any other authentication flow can be picked depending on your use case.
+3. Configure Client API
+
+    Update your `values.yaml` file and add to following parameters to the `clientApi` section:
+
+    ```yaml
+    clientApi:
+      # OIDC provider url.
+      issuerURL: "https://myprovider.example.com"
+
+      # The audience is generally the same as the value of the domain field, however
+      # some issuers like keycloak make the audience the client_id of the application/client.
+      audience: "clienapi.example.com"
+    ```
+
+  The `issuerURL` needs to be set to the url of your provider, eg. `https://accounts.google.com` for google or `https://example.eu.auth0.com/` for Auth0.
+  
+  The `audience` is usually the client apis public URL although for some providers it can also be the client id.
+
+Obtain a token from your OAuth provider that can be used to invoke the Uplink Client API.
+
+The example uses the client credentials grant. Replace the token url, client id and client secret with the values obtained from your identity provider.
+
+```sh
+export IDP_TOKEN_URL="https://myprovider.example.com/token"
+export CLIENT_ID="inlets-uplink"
+export CLIENT_SECRET="$(cat ./client-secret.txt)"
+
+curl -S -L -X POST "${IDP_TOKEN_URL}" \
+  --header 'Content-Type: application/x-www-form-urlencoded' \
+  --data-urlencode "client_id=${CLIENT_ID}" \
+  --data-urlencode "client_secret=${CLIENT_SECRET}" \
+  --data-urlencode 'scope=openid' \
+  --data-urlencode 'grant_type=client_credentials'
+```
+
+Use the token as bearer token in the `Authorization` header when making requests to the API.
+
+```sh
+curl -i \
+  -H "Authorization: Bearer ${TOKEN}" \
+  "$CLIENT_API/v1/tunnels/$NAME?namespace=$NAMESPACE"
+```
+
 ## Tunnels
 
```

Comment:
Introduce that `acmeco` is going to be used as an example namespace. This isn't explained, so isn't clear why it's used in the examples.